docs/proposals/clusterconfig.md

```diff
@@ -0,0 +1,110 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Overview
+
+Proposes a mechanism for specifying cluster-wide runtime config
+for all kube system components via a single config file. This will
+provide an easy way to specify init settings that are shared between
+system components.
+
+## Motivation
+
+Primary motivation is enabling/disabling features that are not tied to
+an API group. API groups can be selectively enabled/disabled in the
+apiserver via the `--runtime-config` commandline flag, but there is
+currently no mechanism to toggle alpha features that are controlled by
+e.g. annotations. This means the burden of controlling whether such
+features are enabled in a particular cluster is on feature implementors;
+they must either define some ad hoc mechanism for toggling (e.g. flag
+on component binary) or else toggle the feature on/off at compile time.
+
+Providing a mechanism for specifying cluster-wide config makes
+deploying non-API features simpler for implementors and cluster admins.
+
+## Design
+
+Cluster-wide runtime config will be specified in a single yaml file
+loaded from a known, default path. TBD whether default config path
+should be overwriteable via flag/env.
+
+Strawman:
+```shell
+$ cat /etc/srv/kubernetes/cluster_config.yaml
+runtimeConfig:
+  enableAppArmor: true
+  enableFooBarBaz: false
```

Comment:
I.e. if you have config maps in the `kube-system` namespace, one for each component of the system, materialized a config maps - is:

```
kubectl get configmaps -o yaml | grep enableAppArmor
```

any harder than also having to reason about two config maps per process type?